fetch/fragments/subresource-redirect.html

```diff
@@ -0,0 +1,48 @@
+<!doctype html>
+<meta charset=utf-8>
+<title>Subresources and fragment preservation</title>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<div id=log></div>
+<!--
+  The source image is 50h x 100w lime green. However, if the viewBox 25,25,50,50 is correctly
+  applied it becomes 25h x 100w. (All images are expected to have this applied per HTTP fragment
+  rules.)
+
+  This image is then drawn on a 50h x 100w transparent black canvas.
+
+  We then test if the top half of the canvas is lime green and the bottom half is transparent black.
+-->
+<img data-desc="Control"
+     src="/images/green.svg#svgView(viewBox(25,25,50,50))">
+<img data-desc="Redirect with the original URL containing a fragment"
+     src=../api/resources/redirect.py?location=/images/green.svg#svgView(viewBox(25,25,50,50))>
+<img data-desc="Redirect with the response Location header containing a fragment"
+     src=../api/resources/redirect.py?location=/images/green.svg%23svgView(viewBox(25,25,50,50))>
+<img data-desc="Redirect with both the original URL and response Location header containing a fragment"
+     src=../api/resources/redirect.py?location=/images/green.svg%23svgView(viewBox(25,25,50,50))#svgView(viewBox(0,0,50,50))>
+<canvas width=100 height=50></canvas>
+<script>
+const ctx = document.querySelector("canvas").getContext("2d"),
+      isColor = (data, r, g, b, a) => {
```

Comment:
This needs to become less precise somehow as it creates intermittent results in Safari. Probably borrow something from canvas tests?